Engine/src/main/java/nullengine/world/WorldCommon.java

```diff
@@ -71,11 +73,11 @@ public void spawnEntity(Entity entity) {
         entityList.add(entity);
     }
 
-    @Deprecated
-    public void playerJoin(Player player) {
-        // FIXME:
+
+    public void onPlayerJoin(@Nonnull Player player) {
+        Player player1=Validate.notNull(player);
```

Comment:
Please use `Objects.requireNonNull`.

You don't need to receive method return value. For example:
```java
public void playerJoin(@Nonnull Player player) {
        Objects.requireNonNull(player); // if player is null, will throw NullPointerException.
        var entity = new PlayerEntity(entityList.size(), this);
        player.controlEntity(entity);
        player.controlEntity(entity);
        spawnEntity(entity);
        players.add(player);
}
```